### ISSUE TITLE
TagGroup doesn't add role attr when collection is empty

### ISSUE THREAD

**Issue by mauron85** [adobe/react-spectrum#8354]
### Provide a general summary of the issue here

When the TagGroup collection is empty, it still renders a `<div>` element with ARIA-related attributes, but the role attribute is missing. This results in a DOM node with ARIA attributes applied to an element that lacks a valid role, which violates accessibility best practices as outlined in [ARIA prohibited attributes](https://accessibilityinsights.io/info-examples/web/aria-prohibited-attr/).

The problematic logic appears to be in the useTagGroup hook:
https://github.com/adobe/react-spectrum/blob/4150364a2dfa249ca149723548c62e0d3325476f/packages/%40react-aria/tag/src/useTagGroup.ts#L108C19-L108C29

### 🤔 Expected Behavior?

If no role is added to tag group element div then don't add other aria attributes aswell or keep role attribute grid even when collection is empty.
 

### 😯 Current Behavior

When TagGroup collection is empty it creates div with all aria attributtes but role is absent.

### 💁 Possible Solution

1. don't add other aria attributes when role is not set
or
3. keep role attribute grid even when collection is empty.

### 🔦 Context

We are checking our application using browserstack accesibility toolkit and it's giving us errors because of TagGroup role issue.

### 🖥️ Steps to Reproduce

1. Create TagGroup with no Tags
2. Observe generated dom element for TagGroup - missing role with there are not Tags

### Version

react-aria: 3.40.0,  react-aria-components: 1.9.0

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

Windows

**Comment by reidbarber** [comment]
I'm not sure keeping the grid role while the collection is empty would be correct:

> A composite [widget](https://w3c.github.io/aria/#widget) containing a collection of one or more rows with one or more cells where some or all cells in the grid are focusable by using methods of two-dimensional navigation, such as directional arrow keys.

https://w3c.github.io/aria/#grid


So it looks like removing the aria-label and aria-labelledby would be the better solution.

**Comment by reidbarber** [comment]
Spoke with @majornista and he recommended experimenting with using a [group](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/group_role) role in this empty state and seeing if that announcement would be better.
